src/ol/render/webgl/VectorStyleRenderer.js

```diff
@@ -321,14 +337,61 @@ class VectorStyleRenderer {
     this.setHelper(helper);
   }
 
+  /**
+   * Will apply the style filter when generating geometry batches (if it can be evaluated outside a map context)
+   * @param {import("../../expr/expression.js").ExpressionValue} filter Style filter
+   * @return {function(import('../../Feature.js').FeatureLike): boolean} Feature filter
+   * @private
+   */
+  computeFeatureFilter(filter) {
+    const parsingContext = newParsingContext();
+    /**
+     * @type {import('../../expr/cpu.js').ExpressionEvaluator}
+     */
+    let compiled;
+    try {
+      compiled = buildExpression(filter, BooleanType, parsingContext);
```

Comment:
I'm still not clear on where we throw due to an expression not being supported on the CPU. I can see where we throw in cases of user error (unexpected syntax in an expression). But maybe the user will see other errors in this case.

Reply:
You can take a look at the test here: https://github.com/openlayers/openlayers/pull/16564/commits/0e5454b15bf8ab5ff62e950b11800616fcdbd27d#diff-504d5d26454652e78f49b181c5e9327a1646f81f9b26144f31e01fec8ad0b076

If the provided filter expression cannot be compiled on the GPU an exception will be thrown and won't be catched by the library code this time.